src/ParserData.php

```diff
@@ -349,18 +349,11 @@ public function pushSemanticDataToParserOutput() {
 	public function markParserOutput() {
 		$this->parserOutput->setTimestamp( wfTimestampNow() );
 
-		if ( method_exists( $this->parserOutput, 'setPageProperty' ) ) {
-			$this->parserOutput->setPageProperty(
-				'smw-semanticdata-status',
-				$this->semanticData->getProperties() !== []
-			);
-		} else {
-			// MW < 1.38
-			$this->parserOutput->setProperty(
-				'smw-semanticdata-status',
-				$this->semanticData->getProperties() !== []
-			);
-		}
+		// ParserOutput::setExtensionData can only take scalar values
+		$this->parserOutput->setExtensionData(
+			'smw-semanticdata-status',
+			$this->semanticData->getProperties() !== [] ? 'true' : 'false'
```

Comment:
Why store `'false'`? I don't see it used anywhere. Also the comment is wrong. Booleans are scalar values. If they don't work, that's probably a bug in MediaWiki, but I see it used, even in SMW itself: https://codesearch.wmcloud.org/search/?q=-%3EsetExtensionData.*%28true%7Cfalse%29&files=&excludeFiles=&repos=

Reply:
Yes you are correct, I misunderstood the [documentation](https://doc.wikimedia.org/mediawiki-core/REL1_39/php/classParserOutput.html#ac15ccba16593e74bd0d500c1dbec5c60) as it didn't explicitly mention boolean. I will update them to use boolean value instead.